File: Persistencia/Ejercicio-1/Empleado.py
class Empleado:
    def __init__(self, nombre: str, edad: int, salario: float):
        self._nombre = nombre
        self._edad = edad
        self._salario = salario
    def get_nombre(self):
        return self._nombre
    def get_edad(self):
        return self._edad
    def get_salario(self):
        return self._salario
    def __str__(self):
        return f"Empleado{{nombre='{self._nombre}', edad={self._edad}, salario={self._salario}}}"

class ArchivoEmpleado:
    def __init__(self, nomA: str):
        self.nomA = nomA
        self.empleados = []
    def crear_archivo(self):
        self.empleados.clear()
    def guardar_empleado(self, e: Empleado):
        self.empleados.append(e)
    def busca_empleado(self, nombre: str):
        for e in self.empleados:
            if e.get_nombre().lower() == nombre.lower():
                return e
        return None
    def mayor_salario(self, salario: float):
        for e in self.empleados:
            if e.get_salario() > salario:
                return e
        return None
    def mostrar_empleados(self):
        for e in self.empleados:
            print(e)

if __name__ == "__main__":
    archivo = ArchivoEmpleado("Empleados2025")

    e1 = Empleado("Carlos", 30, 1500.0)
    e2 = Empleado("Lucia", 28, 2000.0)
    e3 = Empleado("Pedro", 35, 1800.0)

    archivo.guardar_empleado(e1)
    archivo.guardar_empleado(e2)
    archivo.guardar_empleado(e3)

    print("Buscar 'Lucia':")
    encontrado = archivo.busca_empleado("Lucia")
    print(encontrado if encontrado else "No encontrado")

    print("\nEmpleado con salario mayor a 1600:")
    mejor_pagado = archivo.mayor_salario(1600.0)
    print(mejor_pagado if mejor_pagado else "Ninguno encontrado")

File: Persistencia/Ejercicio-1/Empleado.java
import java.util.*;

class ArchivoEmpleado{
    private String nomA;
    private List<Empleado> empleados;

    public ArchivoEmpleado(String n) 
    {
        this.nomA = n;
        this.empleados = new ArrayList<>();
    }

    public void crearArchivo() 
    {
        empleados.clear(); 
    }

    public void guardarEmpleado(Empleado e) 
    {
        empleados.add(e);
    }

    public Empleado buscaEmpleado(String n) 
    {
        for (Empleado e : empleados) 
        {
            if (e.getNombre().equalsIgnoreCase(n))
                return e;
        }
        return null; 
    }

    public Empleado mayorSalario(float s) 
    {
        for (Empleado e : empleados) 
        {
            if (e.getSalario() > s)
                return e;
        }
        return null;
    }

    public void mostrarEmpleados() 
    {
        for (Empleado e : empleados)
            System.out.println(e);
    }
}
class Empleado{
    private String nombre;
    private int edad;
    private float salario;

    public Empleado(String n, int e, float s) 
    {
        this.nombre = n;
        this.edad = e;
        this.salario = s;
    }

    public String getNombre() {
        return nombre;
    }

    public int getEdad() {
        return edad;
    }

    public float getSalario() {
        return salario;
    }

    @Override
    public String toString() 
    {
        return "Empleado{" +"nombre='" + nombre + '\'' +", edad=" + edad +", salario=" + salario +'}';
    }
}
class Main {
    public static void main(String[] args) {
        ArchivoEmpleado archivo = new ArchivoEmpleado("Empleados2025");

        Empleado e1 = new Empleado("Carlos", 30, 1500f);
        Empleado e2 = new Empleado("Lucia", 28, 2000f);
        Empleado e3 = new Empleado("Pedro", 35, 1800f);

        archivo.guardarEmpleado(e1);
        archivo.guardarEmpleado(e2);
        archivo.guardarEmpleado(e3);

        System.out.println("Buscar 'Lucia':");
        Empleado encontrado = archivo.buscaEmpleado("Lucia");
        System.out.println(encontrado != null ? encontrado : "No encontrado");

        System.out.println("\nEmpleado con salario mayor a 1600:");
        Empleado mejorPagado = archivo.mayorSalario(1600f);
        System.out.println(mejorPagado != null ? mejorPagado : "Ninguno encontrado");
    }
}